evap/staff/views.py

```diff
@@ -139,105 +144,152 @@ def __init__(self):
     return render(request, "staff_semester_view.html", template_data)
 
 
+class EvaluationOperation:
+    email_template_name = None
+    confirmation_message = None
+
+    @staticmethod
+    def applicable_to(evaluation):
+        raise NotImplementedError
+
+    @staticmethod
+    def warning_for_unapplicables(amount):
+        return ungettext("{} evaluation can not be updated. It was removed from the selection.",
+            "{} evaluations can not be updated. They were removed from the selection.", amount).format(amount)
+
+    @staticmethod
+    def apply(request, evaluations, email_template=None):
+        raise NotImplementedError
+
+
+class RevertToNewOperation(EvaluationOperation):
+    @staticmethod
+    def applicable_to(evaluation):
+        return evaluation.state in ['prepared', 'editor_approved', 'approved']
+
+    @staticmethod
+    def warning_for_unapplicables(amount):
+        return ungettext("{} evaluation can not be reverted, because it already started. It was removed from the selection.",
+            "{} evaluations can not be reverted, because they already started. They were removed from the selection.", amount).format(amount)
+
+    @staticmethod
+    def confirmation_message():
+        return _("Do you want to revert the following evaluations to preparation?")
```

Comment:
Although I would like this to be a property, I'm not sure how to do that on class level. As it is a translated string, either there needs to be a function behind this property or it's a property like `email_template_name`, but with a lazy ugettext. In my testing, even with how it is now, the strings were not translated, although there should have been a translation available, as these weren't new strings.

Reply:
I guess the `_lazy` translation methods are what you want.

Reply:
I'll try on Monday.